esi_leap/api/controllers/v1/node.py

```diff
@@ -63,19 +64,26 @@ class NodesController(rest.RestController):
     @wsme_pecan.wsexpose(NodeCollection)
     def get_all(self):
         context = pecan.request.context
-        nodes = ironic.get_node_list(context)
+
+        nodes = None
+        project_list = None
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            f1 = executor.submit(ironic.get_node_list, context)
+            f2 = executor.submit(keystone.get_project_list)
+            f3 = executor.submit(offer_obj.Offer.get_all,
+                                 {'status': [statuses.AVAILABLE]}, context)
+            f4 = executor.submit(lease_obj.Lease.get_all,
```

Comment:
I don't think we need to do this for offers and leases, as those are direct database calls, which should be very fast. Ironic/Keystone calls are API calls, so they'll be a lot slower!

Reply:
Ah, okay!